src/fu-util.c

```diff
@@ -957,17 +957,36 @@ fu_util_get_history (FuUtilPrivate *priv, gchar **values, GError **error)
 	return TRUE;
 }
 
+static FwupdDevice*
+fu_util_get_device_or_prompt (FuUtilPrivate *priv, gchar **values, GError **error)
+{
+	FwupdDevice *dev = NULL;
+
+	/* get device to use */
+	if (g_strv_length (values) >= 1) {
+		if (g_strv_length (values) > 1)
+			for (guint i=1; i < g_strv_length (values); i++)
+				g_debug ("Ignoring extra input %s", values[i]);
+		dev = fwupd_client_get_device_by_id (priv->client, values[0],
+						     NULL, error);
+		if (dev != NULL)
+			return dev;
+		g_print ("%s\n", (*error)->message);
```

Comment:
Kinda nasty. A local `GError *error_local = NULL` would be a better way to do this imho.

Reply:
Ok. Will adjust to a local error.